### PR TITLE
Add future annotations as some modules fail with Python 3.9

### PR DESCRIPTION
**Pull request by nabenabe0928**
## Contributor Agreements

Please read the [contributor agreements](https://github.com/optuna/optunahub-registry/blob/main/CONTRIBUTING.md#contributor-agreements) and if you agree, please click the checkbox below.

- [x] I agree to the contributor agreements.

## Motivation

As some packages fail with Python 3.9, I added future annotations and modified the PR template accordingly.

## Description of the changes

- add future annotations
- modify the PR template

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -25,5 +25,6 @@ Otherwise, please check the following TODO list:
 - [ ] Replace `<COPYRIGHT HOLDER>` in `LICENSE` of your package with your name
 - [ ] Fill out `README.md` in your package
 - [ ] Add import statements of your function or class names to be used in `__init__.py`
+- [ ] (Optional) Add `from __future__ import annotations` at the head of any Python files that include MyPy typing to support older Python versions
```
Comment (not522) [inline .github/pull_request_template.md:28]: How about dropping `MyPy`? `from __future__ import annotations` is not just for MyPy.

```suggestion
- [ ] (Optional) Add `from __future__ import annotations` at the head of any Python files that include typing to support older Python versions
```

Comment (nabenabe0928) [reply .github/pull_request_template.md:28]: Sorry for the late response, it completely slipped my mind!
Yes, I agree with you and I will remove it!

**Comment by nabenabe0928** [comment]
@not522 I applied your suggestion, PTAL!!

**Review by not522 — APPROVED** [review]
LGTM!

**Pull request merged**